### ISSUE TITLE
Duplicates in NPO dataset

### ISSUE THREAD

**Issue by cmainov** [Nonprofit-Open-Data-Collective/np-density-dashboard#3]
Hey Jesse,

I am in the process of creating a rodeo dataset that will be used for the spatial grids (detailing distances between NPOs and board members)...I've come across an interesting finding and that is that there are ~7,000 EINs that are duplicated in the NPO IRS 1023 EZ dataset. Importantly, a feature of some of those duplicates is that they have different geocoded locations (see attached image). I am trying to understand this discrepancy in the data and wanted to reach out for your insights since you have worked with these data extensively...could it be that these reflect NPOs moving addresses over the years?
<img width="361" alt="Screen Shot 2022-08-02 at 2 17 47 PM" src="https://user-images.githubusercontent.com/98112670/182446243-97f3a65f-3c61-451b-bec8-4641db5d723b.png">

**Comment by cmainov** [comment]
e.g., see rows 5 and 6 in the output from the image (forgot to mention that above)

**Comment by cmainov** [comment]
<img width="324" alt="Screen Shot 2022-08-02 at 2 24 20 PM" src="https://user-images.githubusercontent.com/98112670/182446606-b9de7830-efbe-4aad-9a77-ef29552f8010.png">


Here is a more detailed look at the data showing the differences across years that I mentioned.

**Comment by cmainov** [comment]
<img width="421" alt="Screen Shot 2022-08-02 at 2 41 16 PM" src="https://user-images.githubusercontent.com/98112670/182449933-da873073-6f73-417d-b5b1-d0dc282dd209.png">

In this output, we have some duplicated EINs with different coordinates but filed in the same year: This sounds like these may reflect organizations that have more than 1 central location? Would love to get your take on this

**Comment by lecy** [comment]
We need to check back to the raw data - do we actually have the same organization filing twice? 

It could be two things - perhaps the geocoding code identified an ambiguous address, so it returned a couple of results? Or its was a non-match at one step so it gets passed to another step (match by PO box or by ZIP code only), and somehow gets added back to the sample twice. 

More likely, if the EIN appears twice in the raw data with two different addresses, it was a resubmission of the application. Usually it is to submit clarifying information, or perhaps they lost their status for failure to file their annual 990 and had to reapply (though that would typically be a couple of years apart, not the same year).

My guess is they submit an update to their application with a new address.

**Comment by lecy** [comment]
**EIN 11111111** seems dubious, though! I would check that one first.

**Comment by lecy** [comment]
For these cases they are likely applying for reinstatement of nonprofit status after failing to file the 990 in a timely manner and having their status revoked (there is a lag between the first and second ruledate): 

https://github.com/Nonprofit-Open-Data-Collective/np-density-dashboard/issues/3#issuecomment-1203070458

**Comment by cmainov** [comment]
Thank you for following up

**Comment by cmainov** [comment]
I agree about EIN 1111111. I will take a look...I think this is something that will need to be addressed. If we are counting the same organization twice, it will obviously skew the results. Just taking note of it for now.

**Comment by cmainov** [comment]
<img width="811" alt="Screen Shot 2022-08-03 at 9 49 32 AM" src="https://user-images.githubusercontent.com/98112670/182624706-095ea16f-8164-4e7f-b743-7ce1d2c4c1b7.png">

here are the three instances of EIN == 1111111...they are all different organizations in different states

**Comment by cmainov** [comment]
<img width="466" alt="Screen Shot 2022-08-03 at 10 02 17 AM" src="https://user-images.githubusercontent.com/98112670/182627682-84c291b8-d42f-4e05-b687-c8d0a741d2d8.png">

SOLVED: Hey Jesse, I found another identifying column in the dataset `Case.Number` (I'm guessing every organization that files is given one of these at the time of filing, but maybe you can fill me in more on this)...It looks like there are only 110 duplicates when I use this column as the row identifier. Upon checking them, the coordinates are the same for the duplicate rows, which tells me that this is the scenario that you posited in your response: namely, that we are dealing with organizations that likely had to submit more information at a later time.

Just to have it documented: I believe we should be using the `Case.Number` as we progress into other realms of analysis/data mgmt
